Setting of preferences in UI test cases

@kittaakos , the UI test DialogUITest testExtractTxtFile sets a preference but the model processor doesn't see it. You won't be able to actually run the test successfully because you don't have the services running, but perhaps you can figure out why the preferences set isn't in the preference store accessed by the model processor?

Which branch? I have compiler errors on the `master`:
I have built it against the SADL `master`.

```
Writing code to "/Users/akos.kitta/git/DARPA-ASKE-TA1/com.ge.research.sadl.darpa.aske.dialog.parent/com.ge.research.sadl.darpa.aske.dialog/../com.ge.research.sadl.darpa.aske.dialog/src-gen/com/ge/research/sadl/darpa/aske/parser/antlr/lexer/jflex/DialogFlexer.java"
19249 [org.eclipse.emf.mwe2.launch.runtime.Mwe2Launcher.main()] INFO  text.xtext.generator.XtextGenerator  - Generating common infrastructure
19269 [org.eclipse.emf.mwe2.launch.runtime.Mwe2Launcher.main()] INFO  .emf.mwe2.runtime.workflow.Workflow  - Done.
[INFO] 
[INFO] --- xtend-maven-plugin:2.15.0:compile (default) @ com.ge.research.sadl.darpa.aske.dialog ---
[ERROR] 
ERROR: 	DialogProcessorContextPreferenceValuesProvider.xtend - /Users/akos.kitta/git/DARPA-ASKE-TA1/com.ge.research.sadl.darpa.aske.dialog.parent/com.ge.research.sadl.darpa.aske.dialog/src/com/ge/research/sadl/darpa/aske/processing/DialogProcessorContextPreferenceValuesProvider.xtend
4: com.ge.research.sadl.processing.IModelProcessor.ProcessorContextPreferenceValuesProvider cannot be resolved to a type.
[ERROR] 
ERROR: 	DialogProcessorContextPreferenceValuesProvider.xtend - /Users/akos.kitta/git/DARPA-ASKE-TA1/com.ge.research.sadl.darpa.aske.dialog.parent/com.ge.research.sadl.darpa.aske.dialog/src/com/ge/research/sadl/darpa/aske/processing/DialogProcessorContextPreferenceValuesProvider.xtend
16: ProcessorContextPreferenceValuesProvider cannot be resolved to a type.
[ERROR] 
ERROR: 	DialogProcessorContextPreferenceValuesProvider.xtend - /Users/akos.kitta/git/DARPA-ASKE-TA1/com.ge.research.sadl.darpa.aske.dialog.parent/com.ge.research.sadl.darpa.aske.dialog/src/com/ge/research/sadl/darpa/aske/processing/DialogProcessorContextPreferenceValuesProvider.xtend
26: The method or field preferenceProvider is undefined
[ERROR] 
ERROR: 	DialogProcessorContextPreferenceValuesProvider.xtend - /Users/akos.kitta/git/DARPA-ASKE-TA1/com.ge.research.sadl.darpa.aske.dialog.parent/com.ge.research.sadl.darpa.aske.dialog/src/com/ge/research/sadl/darpa/aske/processing/DialogProcessorContextPreferenceValuesProvider.xtend
42: The method or field preferenceProvider is undefined
[ERROR] 
ERROR: 	DialogProcessorContextPreferenceValuesProvider.xtend - /Users/akos.kitta/git/DARPA-ASKE-TA1/com.ge.research.sadl.darpa.aske.dialog.parent/com.ge.research.sadl.darpa.aske.dialog/src/com/ge/research/sadl/darpa/aske/processing/DialogProcessorContextPreferenceValuesProvider.xtend
46: The method or field super is undefined
[ERROR] 
ERROR: 	DialogProcessorContextPreferenceValuesProvider.xtend - /Users/akos.kitta/git/DARPA-ASKE-TA1/com.ge.research.sadl.darpa.aske.dialog.parent/com.ge.research.sadl.darpa.aske.dialog/src/com/ge/research/sadl/darpa/aske/processing/DialogProcessorContextPreferenceValuesProvider.xtend
16: Superclass must be a class
[ERROR] 
ERROR: 	DialogProcessorContextPreferenceValuesProvider.xtend - /Users/akos.kitta/git/DARPA-ASKE-TA1/com.ge.research.sadl.darpa.aske.dialog.parent/com.ge.research.sadl.darpa.aske.dialog/src/com/ge/research/sadl/darpa/aske/processing/DialogProcessorContextPreferenceValuesProvider.xtend
20: The method getPreferenceValues(Resource) of type DialogProcessorContextPreferenceValuesProvider must override a superclass method.
[ERROR] 
ERROR: 	DialogRuntimeModule.xtend - /Users/akos.kitta/git/DARPA-ASKE-TA1/com.ge.research.sadl.darpa.aske.dialog.parent/com.ge.research.sadl.darpa.aske.dialog/src/com/ge/research/sadl/darpa/aske/DialogRuntimeModule.xtend
44: com.ge.research.sadl.processing.IModelProcessor.ProcessorContextPreferenceValuesProvider cannot be resolved to a type.
[ERROR] 
```

--- a/com.ge.research.sadl.darpa.aske.dialog.parent/com.ge.research.sadl.darpa.aske.dialog.ui.tests/src/com/ge/research/sadl/darpa/aske/dialog/ui/tests/AbstractDialogPlatformTest.xtend
+++ b/com.ge.research.sadl.darpa.aske.dialog.parent/com.ge.research.sadl.darpa.aske.dialog.ui.tests/src/com/ge/research/sadl/darpa/aske/dialog/ui/tests/AbstractDialogPlatformTest.xtend
@@ -1,30 +1,33 @@
 package com.ge.research.sadl.darpa.aske.dialog.ui.tests
 
+import com.ge.research.sadl.darpa.aske.ui.tests.DialogUiInjectorProvider
+import com.ge.research.sadl.jena.IJenaBasedModelProcessor
+import com.ge.research.sadl.model.gp.Rule
+import com.ge.research.sadl.model.gp.SadlCommand
+import com.ge.research.sadl.tests.SadlTestAssertions
+import com.ge.research.sadl.ui.tests.AbstractSadlPlatformTest
 import com.hp.hpl.jena.ontology.OntModel
 import java.util.List
 import org.eclipse.emf.ecore.resource.Resource
 import org.eclipse.xtext.resource.XtextResource
+import org.eclipse.xtext.testing.InjectWith
+import org.eclipse.xtext.testing.XtextRunner
 import org.eclipse.xtext.validation.Issue
-import com.ge.research.sadl.ui.tests.AbstractSadlPlatformTest
-import com.ge.research.sadl.model.gp.SadlCommand
-import com.ge.research.sadl.model.gp.Rule
-import com.ge.research.sadl.jena.IJenaBasedModelProcessor
-import com.ge.research.sadl.tests.SadlTestAssertions
-import com.ge.research.sadl.darpa.aske.tests.AbstractDialogTest
+import org.junit.runner.RunWith
 
+@RunWith(XtextRunner)
+@InjectWith(DialogUiInjectorProvider)
 abstract class AbstractDialogPlatformTest extends AbstractSadlPlatformTest {
-	
-	
+
 	protected def Resource assertValidatesDialogTo(Resource resource,
 		(OntModel, List<Rule>, List<SadlCommand>, List<Issue>, IJenaBasedModelProcessor)=>void assertions) {
 
 		return SadlTestAssertions.assertValidatesTo(resource as XtextResource, assertions);
 	}
-	
+
 	protected def Resource assertValidatesDialogTo(Resource resource, boolean rawResults, boolean treatAsConclusion,
 		(OntModel, List<Object>, List<SadlCommand>, List<Issue>, IJenaBasedModelProcessor)=>void assertions) {
-			
 //			return AbstractDialogTest.assertValidatesTo(resource as XtextResource, assertions); 
-		}
+	}
 
 }

